packages/office-ui-fabric-react/src/components/Slider/Slider.base.tsx

```diff
@@ -80,31 +99,93 @@ export class SliderBase extends React.Component<ISliderProps, ISliderState> impl
       showValue,
       buttonProps,
       vertical,
-      valueFormat,
       styles,
       theme,
       originFromZero,
+      ranged,
     } = this.props;
     const value = this.value;
     const renderedValue = this.renderedValue;
-    const thumbOffsetPercent: number = min === max ? 0 : ((renderedValue! - min!) / (max! - min!)) * 100;
-    const zeroOffsetPercent: number = min! >= 0 ? 0 : (-min! / (max! - min!)) * 100;
+    const renderedLowerValue = this.renderedLowerValue;
+    const thumbOffsetPercent: number = this._getPercent(renderedValue);
+    const lowerThumbOffsetPercent: number = this._getPercent(renderedLowerValue);
+    const originValue = originFromZero ? 0 : min;
+
+    const originPercent = this._getPercent(originValue);
+    const activeSectionWidth = ranged
+      ? thumbOffsetPercent - lowerThumbOffsetPercent
+      : Math.abs(originPercent - thumbOffsetPercent);
+    const topSectionWidth = Math.min(100 - thumbOffsetPercent, 100 - originPercent);
+    const bottomSectionWidth = ranged ? lowerThumbOffsetPercent : Math.min(thumbOffsetPercent, originPercent);
+
     const lengthString = vertical ? 'height' : 'width';
     const onMouseDownProp: {} = disabled ? {} : { onMouseDown: this._onMouseDownOrTouchStart };
     const onTouchStartProp: {} = disabled ? {} : { onTouchStart: this._onMouseDownOrTouchStart };
     const onKeyDownProp: {} = disabled ? {} : { onKeyDown: this._onKeyDown };
+    const onFocusProp: {} = disabled ? {} : { onFocus: this._onThumbFocus };
+
     const classNames = getClassNames(styles, {
       className,
       disabled,
       vertical,
-      showTransitions: renderedValue === value,
+      showTransitions: renderedValue === value || renderedLowerValue === this.lowerValue,
```

Comment:
Make sure that any modifications to the behavior are guarded by a check that `ranged` is true.
```suggestion
      showTransitions: renderedValue === value || (ranged && renderedLowerValue === this.lowerValue),
```